dashboard/src/reducers/charts.ts

```diff
@@ -70,14 +71,15 @@ const chartsReducer = (
         state.records.set(state.page, true); // mark the current page as successfully retrieved
         state.records.set(state.page + 1, false); // mark the next page as pending (not undefined)
       }
+      const isLastPage = action.payload.page >= action.payload.totalPages;
```

Comment:
shouldn't this be `page === totalPages`? (not sure why the `>=`, if that's possible.)

Reply:
Yes, in theory, we reach the end when the `page===totalPages`. It was just to cover the edge case in which two `receiveCharts` arrive at the same time and, "lastPage" and "lastPage+1" are both marked to be fetched. So I directly consider `lastPage+n` as the same case.

However, it should occur and, if it does, we can just ignore subsequents `lastPage+n` request.